omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/PasswordResetController2_2.java

```diff
@@ -11,39 +11,48 @@
 
 import java.util.Map;
 
-import org.openmrs.User;
 import org.openmrs.api.InvalidActivationKeyException;
 import org.openmrs.api.UserService;
 import org.openmrs.api.ValidationException;
+import org.openmrs.api.context.Daemon;
+import org.openmrs.module.DaemonToken;
+import org.openmrs.module.DaemonTokenAware;
 import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.Task.PasswordResetTask;
 import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
 import org.openmrs.notification.MessageException;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
+@Component("ResetPasswordComponent")
 @Controller
 @RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/passwordreset")
-public class PasswordResetController2_2 extends BaseRestController {
-	
+public class PasswordResetController2_2 extends BaseRestController  implements DaemonTokenAware {
```

Comment:
Why did you make this `DaemonTokenAware`?

Reply:
Here are more examples for you to look at: https://github.com/search?l=&q=org%3Aopenmrs+DaemonTokenAware&ref=advsearch&type=Code